<launch>
    <!-- 0 -> statinary.bag; 1 -> with_gripper_perpendicular.bag; 2 -> with_gripper_parallel.bag -->
    <!-- 3 -> self_intersection.bag; 4 -> folding -->
    <arg name="bag_file" default="0" />
    <arg name="use_eval_rope" default="true" />
    <arg name="double_nodes" default="false" />
    <arg name="compute_srt" default="true" />

    <node name="bcpd" pkg="bcpd-dlo-tracking" type="bcpd" output="screen">
        <param name="beta_1" value="5" />
        <param name="beta_2" value="5" />
        <param name="lambda" value="40" />
        <param name="tao" value="0.995" />

        <param name="omega" value="0.0" />
        <param name="kappa" type="double" value="1e16" />
        <param name="gam" value="1.1" />
        <param name="zeta" type="double" value="1e-8" />

        <param name="max_iter" value="50" />
        <param name="tol" value="0.0001" />
        <param name="use_prev_sigma2" value="true" />
        
        <param name="use_eval_rope" value="$(arg use_eval_rope)" />
        <param name="bag_file" value="$(arg bag_file)" />
        <param name="num_of_nodes" value="40" />
        <param name="downsample_leaf_size" value="0.005" />

        <param name="double_nodes" value="$(arg double_nodes)" />
        <param name="compute_srt" value="$(arg compute_srt)" />
    </node>

</launch>